### PR TITLE
feat(user-notification): Seen and read and senderId

### PR DESCRIPTION
**Pull request by rafnarnason**
Changing data structure
dropping status in favour of seen and read booleans
adding senderId property
new endpoints for read seen counts and mark-all-as-seen

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/services/user-notification/src/app/modules/notifications/dto/notification.dto.ts [hunk 1]
```diff
@@ -81,6 +84,10 @@ export class RenderedNotificationDto {
   @IsUUID()
   messageId!: string
 
+  @ApiProperty({ example: 'Þjóðskrá' })
+  @IsString()
+  senderId!: string | null | undefined
```
Comment (kksteini) [inline apps/services/user-notification/src/app/modules/notifications/dto/notification.dto.ts:89]: `!`
`undefined`

It's definitely a string (or null or undefined) :laughing: 

---

Isn't this a bit strange?

File: apps/services/user-notification/src/app/modules/notifications/notifications.service.ts [hunk 2]
```diff
@@ -341,4 +341,48 @@ export class NotificationsService {
...
+      this.logger.error('Error getting unseen notifications count:', error)
+      throw new InternalServerErrorException('Error getting unseen notifications count')
+    }
+  }
+
+  async markAllAsSeen(user: User): Promise<void> {
+    try {
+      await this.notificationModel.update(
+        { seen: true },
+        { where: { recipient: user.nationalId, seen: false } }
```
Comment (kksteini) [inline apps/services/user-notification/src/app/modules/notifications/notifications.service.ts:379]: I'm not 100% certain how `CreateOptional` works but as `seen` is declared that way, is it then possible that `null` entries for `seen` exist?

Should this be something like `{ where: { recipient: user.nationalId, seen: { $not: true } }`? (I don't remember the ORM too well)

**Review by kksteini — APPROVED** [review]

File: apps/services/user-notification/src/app/modules/notifications/dto/notification.dto.ts (hunk 1 above)
Comment (rafnarnason) [reply apps/services/user-notification/src/app/modules/notifications/dto/notification.dto.ts]: fixed

**Review by thordurhhh — APPROVED** [review]
🙌

**Pull request merged**